sympy/parsing/c/c_parser.py

```diff
@@ -308,12 +337,12 @@ def transform_var_decl(self, node):
                     or child.kind == cin.CursorKind.PAREN_EXPR
                     or child.kind == cin.CursorKind.UNARY_OPERATOR
                     or child.kind == cin.CursorKind.CXX_BOOL_LITERAL_EXPR):
-                    if (node.type.kind == cin.TypeKind.INT):
-                        type = IntBaseType(String('integer'))
-                    elif (node.type.kind == cin.TypeKind.FLOAT):
-                        type = FloatBaseType(String('real'))
-                    elif (node.type.kind == cin.TypeKind.BOOL):
-                        type = Type(String('bool'))
+                    if node.type.kind in self._data_types["int"]:
+                        type = self._data_types["int"][node.type.kind]
+                    elif node.type.kind in self._data_types["float"]:
+                        type = self._data_types["float"][node.type.kind]
+                    elif node.type.kind in self._data_types["bool"]:
+                        type = self._data_types["bool"][node.type.kind]
```

Comment:
I feel it might be better to move all the type declarations to before the try statement, It'll help reduce a lot of the redundancy.
Just keep the value declarations here

Reply:
Looks like, it will indeed reduce redundanies, working on it now!